oomph/c_output.py

```diff
@@ -216,15 +222,22 @@ def emit_funcdef(self, funcdef: tast.FuncDef, c_name: str) -> str:
         )
         body_statements = "".join(self.emit_statement(s) for s in funcdef.body)
 
-        if funcdef.type.returntype is not None:
+        if isinstance(funcdef, tast.FuncDef):
+            assert isinstance(funcdef.var.type, FunctionType)
+            functype = funcdef.var.type
+        else:
+            functype = funcdef.type
+
+        if functype.returntype is not None:
             self.before_body += (
-                f"{self.file_emitter.emit_type(funcdef.type.returntype)} retval;"
+                f"{self.file_emitter.emit_type(functype.returntype)} retval;"
             )
             self.after_body += "return retval;"
 
+        varnames = [self.variable_names[var] for var in funcdef.argvars]
         return f"""
-        {self.file_emitter.emit_type(funcdef.type.returntype)}
-        {c_name}({arg_declarations or "void"})
+        {"static" if static else ""}
+        {self.file_emitter.declare_function(c_name, functype, varnames)}
```

Comment:
Must change `declare_function`, but this might be nice:

```suggestion
        {self.file_emitter.declare_function(c_name, functype, varnames, static)}
```